flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolderTest.java

```diff
@@ -280,6 +281,36 @@ public void takeCheckpointAfterAbortedCheckpoint() throws Exception {
 		));
 	}
 
+	@Test
+	public void testFailingJobMultipleTimesNotCauseCascadingJobFailure() throws Exception {
+		Function<OperatorCoordinator.Context, OperatorCoordinator> coordinatorProvider =
+			context -> new TestingOperatorCoordinator(context) {
+				@Override
+				public void handleEventFromOperator(int subtask, OperatorEvent event) {
+					context.failJob(new RuntimeException("Artificial Exception"));
+				}
+			};
+		final TestEventSender sender = new TestEventSender();
+		final OperatorCoordinatorHolder holder = createCoordinatorHolder(sender, coordinatorProvider);
+
+		holder.handleEventFromOperator(0, new TestOperatorEvent());
+		assertNotNull(globalFailure);
+		final Throwable firstGlobalFailure = globalFailure;
+
+		holder.handleEventFromOperator(1, new TestOperatorEvent());
+		assertEquals("The global failure should be the same instance because the context"
```

Comment:
Nit: I think we want referential equality here (`assertSame`) rather than semantical equality (`assertEquals`).